OpenSim/Simulation/StatesTrajectory.cpp

```diff
@@ -143,16 +143,21 @@ TimeSeriesTable StatesTrajectory::exportToTable(const Model& model,
             requestedStateVars;
     table.setColumnLabels(stateVars);
     size_t numDepColumns = stateVars.size();
-
+    
     // Fill up the table with the data.
     for (size_t itime = 0; itime < getSize(); ++itime) {
         const auto& state = get(itime);
         TimeSeriesTable::RowVector row(static_cast<int>(numDepColumns));
 
         // Get each state variable's value.
-        for (unsigned icol = 0; icol < numDepColumns; ++icol) {
-            row[static_cast<int>(icol)] = 
-                model.getStateVariableValue(state, stateVars[icol]);
+        if (requestedStateVars.empty()) {
+            // This is *much* faster than getting the values one-by-one.
```

Comment:
Hmm...interesting.. 

> though I guess it depends on how many values the user has requested.

I suppose I could do some testing to determine the number of state variables at which to switch schemes.